packages/workflow/src/workflow.ts

```diff
@@ -1119,19 +1137,122 @@ export function defineQuery<Ret, Args extends any[] = [], Name extends string =
+//    the Signal/Update will be executed before Workflow code is executed. If it
+//    is not, then the Signal/Update calls is pushed to a buffer.
+//
+// 2. On each call to setHandler for a given Signal/Update, we make a pass
+//    through the buffer list. If a buffered job is associated with the just-set
+//    handler, then the job is removed from the buffer and the initial
+//    synchronous portion of the handler is invoked on that input (i.e.
+//    preempting workflow code).
+//
+// Thus in the case of Typescript the questions above are answered as follows:
+//
+// 1. A handler is "available" if it is set at the start of the Activation or
+//    becomes set at any point during the Activation. If the handler is not set
+//    initially then it is executed as soon as it is set. Subsequent deletion or
+//    replacement by a different handler has no impact because the jobs it was
+//    handling have already been handled and are no longer in the buffer.
+//
+// 2. The handler is executed as soon as it becomes available. I.e. if the
+//    handler is set at the start of the Activation then it is executed when
+//    first attempting to process the Signal/Update job; alternatively, if it is
+//    set by a setHandler call made by Workflow code, then it is executed as
+//    part of that call (preempting Workflow code). Therefore, a user can rely
+//    on Signal/Update side effects being reflected in e.g. the Workflow return
+//    value, and in the value passed to Continue-As-New. Activation jobs are
+//    processed in the order supplied by sdk-core, i.e. Signals, then Updates,
+//    then other jobs. Within each group, the order sent by the server is
+//    preserved. If the handler is async, it is executed up to its first yield
+//    point.
+//
+// 3. Signal case: If a handler does not become available for a Signal job then
+//    the job remains in the buffer. If a handler for the Signal becomes
+//    available in a subsequent Activation (of the same or a subsequent WFT)
+//    then the handler will be executed. If not, then the Signal will never be
+//    responded to and this causes no error.
+//
+//    Update case: If a handler does not become available for an Update job then
+//    the Update is rejected at the end of the Activation. Thus, if a user does
+//    not want an Update to be rejected for this reason, then it is their
+//    responsibility to ensure that their application and workflow code interact
+//    such that a handler is available for the Update during any Activation
+//    which might contain their Update job. (Note that the user often has
+//    uncertainty about which WFT their Signal/Update will appear in. For
+//    example, if they call startWorkflow() followed by startUpdate(), then they
+//    will typically not know whether these will be delivered in one or two
+//    WFTs. On the other hand there are situations where they would have reason
```

Comment:
I think if they use the UpdateWithStart that is coming that would guarantee one WFT. But there will not be a way to run handlers before everything else since setHandler would need to be called in the first task of UpdateWithStart...